### PR TITLE
feat(data-access): GeoExperiment impact-measurement phases + insights/impactTaskId fields

### PR DESCRIPTION
**Pull request by ssilare-adobe**
## What
Adds the shared-model changes for the **impact-measurement** workflow (see the *Impact Validation Engine | Impact Measurement Workflow* wiki):

- Two new `GeoExperiment.PHASES`: `impact_measurement_started`, `impact_measurement_done`.
- New nullable `insights` attribute — the measurement output (ExperimentInsights schema), stored beside `metadata`.
- New nullable `impactTaskId` attribute — the Mystique task handle, a first-class field like `pre/postScheduleId`, so the engine can resume polling across cron ticks.
- Updated `index.d.ts` types and unit tests.

## Why
The experimentation engine keeps an experiment `IN_PROGRESS` after post-analysis, triggers a Mystique analysis, polls it, and writes the resulting insights back onto the experiment. These fields/phases are the data-model foundation for that.

## Notes
- DynamoDB/electrodb — no SQL migration in this repo.
- Unit tests: 2370 passing; 100% coverage on the changed geo-experiment files; lint clean.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by MysticatBot — APPROVED** [review]
Hey @ssilare-adobe,

> ⚠ Degraded review - no spec document was found for this change (searched the PR links, the touched repos' docs, the architecture/guidelines docs, and linked Jira). This review covers code-level quality but could not validate the change against an agreed design, so confidence is reduced. Add a spec link (PR template section 4) and re-request review for a full-confidence pass.

**Verdict: Approve** - clean additive change following established entity patterns exactly.
**Complexity: LOW** - small diff, single service.
_Changes: Adds impact-measurement lifecycle phases and two new nullable fields (impactTaskId, insights) to the GeoExperiment data model (4 files)._

<details>
<summary>Non-blocking (3): minor issues and suggestions</summary>

- suggestion: `getInsights()` return type could use a named `ExperimentInsights` interface instead of bare `object`, since the PR description documents a known schema shape (version, generatedAt, analyses) - `packages/spacecat-shared-data-access/src/models/geo-experiment/index.d.ts:38`. Matches existing `metadata`/`error` convention, so optional.
- suggestion: Add a test that sets `insights` to `null` to confirm the nullable path works through the model layer - `packages/spacecat-shared-data-access/test/unit/models/geo-experiment/geo-experiment.model.test.js:178`
- suggestion: Add a negative validation test for `impactTaskId` with empty string to confirm the `hasText` validator rejects it - `packages/spacecat-shared-data-access/test/unit/models/geo-experiment/geo-experiment.model.test.js:71`

</details>

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 1m 8s | <b>Cost:</b> $4.00 | <b>Commit:</b> <code>e8ed764f95c6fd77795015cafeabb3f171b4f398</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

**Review by dipratap — APPROVED** [review]

File: packages/spacecat-shared-data-access/src/models/geo-experiment/geo-experiment.schema.js [hunk 1]
```diff
@@ -34,6 +34,10 @@ const schema = new SchemaBuilder(GeoExperiment, GeoExperimentCollection)
     type: 'string',
     validate: (value) => !value || hasText(value),
   })
+  .addAttribute('impactTaskId', {
```
Comment (dipratap) [inline packages/spacecat-shared-data-access/src/models/geo-experiment/geo-experiment.schema.js]: we can save the impactTaskId as a new field in experiment metadata. Creating a new field would not be required here.

**Review by dipratap — CHANGES_REQUESTED** [review]

File: packages/spacecat-shared-data-access/src/models/geo-experiment/geo-experiment.schema.js [hunk 2]
```diff
@@ -81,6 +85,10 @@ const schema = new SchemaBuilder(GeoExperiment, GeoExperimentCollection)
     type: 'any',
     validate: (value) => !value || isObject(value),
   })
+  .addAttribute('insights', {
```
Comment (dipratap) [inline packages/spacecat-shared-data-access/src/models/geo-experiment/geo-experiment.schema.js]: We can save the S3 link for the insights S3 location here like we do for the prompts location. The variable needs renaming - insights_location maybe?. The /geo-experiment API, when called with a query param includeInsights=true, will read from S3 and share the measurement metrics like it do for prompts.

Please note that changes to Post-gres table schema require DB migration - https://github.com/adobe/mysticat-data-service. Since insights will have no values for experiments prior to this change so this column should accept null values.

File: packages/spacecat-shared-data-access/src/models/geo-experiment/geo-experiment.schema.js (hunk 1 above)
Comment (dipratap) [reply packages/spacecat-shared-data-access/src/models/geo-experiment/geo-experiment.schema.js]: Lets rename it to impactMeasurementTaskId

Comment (ssilare-adobe) [reply packages/spacecat-shared-data-access/src/models/geo-experiment/geo-experiment.schema.js]: Done in 0f5d486d — removed the `impactTaskId` schema attribute; the task id is now stored in `metadata` under the new `METADATA_KEYS.IMPACT_MEASUREMENT_TASK_ID` (`impactMeasurementTaskId`).

Comment (ssilare-adobe) [reply packages/spacecat-shared-data-access/src/models/geo-experiment/geo-experiment.schema.js]: Done in 0f5d486d — it's now `impactMeasurementTaskId`, stored in `metadata` (no schema column).

File: packages/spacecat-shared-data-access/src/models/geo-experiment/geo-experiment.schema.js (hunk 2 above)
Comment (ssilare-adobe) [reply packages/spacecat-shared-data-access/src/models/geo-experiment/geo-experiment.schema.js]: Done in 0f5d486d — replaced the `insights` object column with a nullable `insightsLocation` string (S3 key, mirroring `promptsLocation`). The insights JSON lives in S3; the /geo-experiment API can read it on `includeInsights=true`. The Postgres migration in mysticat-data-service (nullable column) is tracked as a follow-up in that repo.

**Review by dipratap — APPROVED** [review]
@ssilare-adobe Please also create DB migration PR in mysticat and validate on dev

**Review by MysticatBot — APPROVED** [review]
Hey @ssilare-adobe,

> ⚠ Degraded review - no spec document was found for this change (searched the PR links, the touched repos' docs, the architecture/guidelines docs, and linked Jira). This review covers code-level quality but could not validate the change against an agreed design, so confidence is reduced. Add a spec link (PR template section 4) and re-request review for a full-confidence pass.

**Verdict: Approve** - clean additive change addressing all reviewer feedback, following established patterns.
**Complexity: LOW** - small diff, single service.
_Changes: Adds impact-measurement lifecycle phases, a metadata key for the task ID, and an insightsLocation string field to the GeoExperiment data model (4 files)._

<details>
<summary>Non-blocking (2): minor issues and suggestions</summary>

- suggestion: Update the PR description to match the final implementation - it still references `impactTaskId` as a "first-class field like pre/postScheduleId" and `insights` as "the measurement output", but the code now stores the task ID in metadata (`METADATA_KEYS.IMPACT_MEASUREMENT_TASK_ID`) and uses `insightsLocation` (an S3 pointer, mirroring `promptsLocation`). The inline discussion makes this clear, but the description will confuse future readers landing via git blame.
- nit: `insightsLocation` validator `(value) => !value || hasText(value)` accepts empty string because `!''` is truthy - consistent with `promptsLocation` and the rest of the codebase, so not a regression, but worth knowing if downstream consumers assume non-empty.

</details>

## Previously flagged, now resolved
- `impactTaskId` moved from schema attribute to `METADATA_KEYS` constant (per dipratap's request)
- `insights` object replaced with `insightsLocation` S3 pointer string (per dipratap's request)
- TypeScript declarations updated to match

---

<sub><b>Skill:</b> pr-review | <b>Model:</b> us.anthropic.claude-opus-4-6-v1[1m] | <b>Duration:</b> 1m 11s | <b>Cost:</b> $2.50 | <b>Commit:</b> <code>dfc8ec7842ca169a567b02066f3a84a847eee636</code><br>If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>
